mcstatus/responses.py

```diff
@@ -340,3 +354,126 @@ def version(self) -> str:
             Will be removed 2023-12, use :attr:`.name` instead.
         """
         return self.name
+
+
+@dataclass(frozen=True)
+class QueryResponse:
+    """The response object for :meth:`JavaServer.query() <mcstatus.server.JavaServer.query>`."""
+
+    raw: RawQueryResponse
+    """Raw response from the server.
+
+    This is :class:`~typing.TypedDict` actually, please see sources to find what is here.
+    """
+    motd: Motd
+    """The MOTD of the server. Also known as description.
+
+    .. seealso:: :doc:`/api/motd_parsing`.
+    """
+    map_name: str
+    """The name of the map. Default is ``world``."""
+    players: QueryPlayers
+    """The players information."""
+    software: QuerySoftware
+    """The software information."""
+    ip: str
+    """The IP address the server is listening/was contacted on."""
+    port: int
+    """The port the server is listening/was contacted on."""
+    game_type: str = "SMP"
+    """The game type of the server. Hardcoded to ``SMP`` (survival multiplayer)."""
+    game_id: str = "MINECRAFT"
+    """The game ID of the server. Hardcoded to ``MINECRAFT``."""
+
+    @classmethod
+    def build(cls, raw: RawQueryResponse, players_list: list[str]) -> Self:
+        return cls(
+            raw=raw,
+            motd=Motd.parse(raw["hostname"], bedrock=False),
+            map_name=raw["map"],
+            players=QueryPlayers.build(raw, players_list),
+            software=QuerySoftware.build(raw["version"], raw["plugins"]),
+            ip=raw["hostip"],
+            port=int(raw["hostport"]),
+            game_type=raw["gametype"],
+            game_id=raw["game_id"],
+        )
+
+    @property
+    @deprecated(replacement="map_name", date="2025-08")
+    def map(self) -> str | None:
+        """
+        .. deprecated:: 11.0.0
```

Comment:
Whoops, missed this, the version also needs changing, we're already at v11, so it should be v12 probably.